liblouisutdml/readconfig.c

```diff
@@ -852,8 +853,12 @@ compileConfig (lbu_FileInfo * nested)
 	    }
 	  break;
 	case 13:
-	  if ((k = checkValues (nested, hyphenationModes)) != NOTFOUND)
+	  if ((k = checkValues (nested, hyphenationModes)) != NOTFOUND) {
 	    ud->hyphenate = k;
+	    if (ud->hyphenate == 1)
+	      // "yes" -> combination of "yes" and "pre"
+	      ud->hyphenate += 2;
```

Comment:
This seems weird. I presume that hyphenate can be 0 (not hyphenated), 1 (hyphenated) or 2 (pre-hyphenated). Could it not become 3 with above code?

Frankly I think the hyphenate int should be an enum.

Reply:
It _will_ become 3. I did this so that I could keep the old behavior of "yes" as "2". The new behavior is kind of a combination of "2" and "1", hence the "3". The old behavior of "yes" can currently not be selected, but we could enable it some day if we want, by adding a new option.

Reply:
Hm, seems like all the more reason to turn this into an enum.

OTOH we don't want to spend too much time on this. But if you'll look at this two years from now you'll never remember these states and magic numbers.

Reply:
I'm hoping I won't ever have to look at it again :)

Reply:
I'm fine with the reasoning you mentioned on the phone